sdks/v4-sdk/src/PositionManager.ts

```diff
@@ -284,9 +282,11 @@ export abstract class V4PositionManager {
 
     let value: string = toHex(0)
 
+    let needToSendEth = isMint(options) && options.migrateOptions?.neededCurrency?.isNative
+
     // If migrating, we need to settle and sweep both currencies individually
-    if (isMint(options) && options.migrate) {
-      if (options.useNative) {
+    if (isMint(options) && options.migrateOptions?.migrate) {
+      if (options.useNative && !needToSendEth) {
```

Comment:
in the case where neededCurrency is ETH and the position is in WETH this path should still be taken i think?

Reply:
I think the opposite right? it would need to wrap not unwrap